clang/lib/Driver/ToolChains/HLSL.cpp

```diff
@@ -506,6 +510,20 @@ HLSLToolChain::TranslateArgs(const DerivedArgList &Args, StringRef BoundArch,
       A->claim();
       continue;
     }
+    if (A->getOption().getID() == options::OPT_dxc_col_major) {
+      DAL->AddSeparateArg(nullptr,
```

Comment:
[`fmatrix_memory_layout_EQ`](https://github.com/llvm/llvm-project/blob/4e05d702f02aea708e8431061ad886c5cc9f4f40/clang/include/clang/Options/Options.td#L4711) is specified as a `Joined` option, so I am surprised this is working as expected. I would have expected it to be:
```suggestion
      DAL->AddJoinedArg(nullptr,
```

Maybe it is because [this code](https://github.com/llvm/llvm-project/blob/4e05d702f02aea708e8431061ad886c5cc9f4f40/clang/lib/Driver/ToolChains/Clang.cpp#L5708) patches it to be a joined arg anyway.